### PR TITLE
5375 bugfix embedded entity edit

### PR DESCRIPTION
**Pull request by LukasStordeur**
# Description

Enabling the possibility to add new nested entities that are within embedded entities. 

closes *#5375*

https://github.com/inmanta/web-console/assets/44098050/d8f2015e-be7e-4737-9a5c-c661f8f07e76

### CODE REVIEW

File: src/UI/Components/ServiceInstanceForm/ServiceInstanceForm.test.tsx [hunk 1]
```diff
@@ -24,6 +24,9 @@ const setup = (
     | EnumField
   )[],
   func: undefined | jest.Mock = undefined,
+  isEdit = false,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  originalAttributes: any = undefined,
```
Comment (matborowczyk) [inline src/UI/Components/ServiceInstanceForm/ServiceInstanceForm.test.tsx:29]: I know that's not much better than any but if I recall correctly for attributes we've got `InstanceAttributeModel` which is still a little enigmatic but we don't need to overwrite eslint

File: src/UI/Components/ServiceInstanceForm/ServiceInstanceForm.tsx [hunk 2]
```diff
@@ -88,15 +121,26 @@ export const ServiceInstanceForm: React.FC<Props> = ({
         });
       }
     },
-    [dirtyInputs],
+    [isDirty],
   );
 
+  /**
```
[the remark this answers is not in the document]
Comment (LukasStordeur) [reply src/UI/Components/ServiceInstanceForm/ServiceInstanceForm.tsx:127]: Just trying to be consistent and add docstrings for all methods I encounter. 
Some might seem irrelevant, but it's subjective. All comments are irrelevant when you know the code by heart. 
And adding docstrings enhances the intellisense too. This is just me picking up old habits to try to document all I encounter.

[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Components/ServiceInstanceForm/ServiceInstanceForm.tsx:127]: Make sense!

**Review by matborowczyk — APPROVED** [review]

File: src/UI/Components/ServiceInstanceForm/ServiceInstanceForm.test.tsx (hunk 1 above)
Comment (LukasStordeur) [reply src/UI/Components/ServiceInstanceForm/ServiceInstanceForm.test.tsx]: updated

**Pull request closed without merging**